doc/melange.rst

```diff
@@ -0,0 +1,296 @@
+  can be used. For example
+  ``(runtime_deps ./path/to/file.css (glob_file_rec ./fonts/*))``.
+
+- ``(emit_stdlib <bool>)`` allows the user to specify whether the Melange
+  standard library should be included as a dependency of the stanza or not. The
+  default is ``true``.
+
+- ``(promote <options>)`` allows promoting the generated ``.js`` files to the
+  source tree. The options are the same as for the :ref:`rule promote mode <promote>`.
+  Adding ``(promote (until-clean))`` to a ``melange.emit`` stanza will cause
+  Dune to copy the ``.js`` files to the source tree and use ``dune clean`` to
+  delete them.
+
+- ``(preprocess <preprocess-spec>)`` specifies how to preprocess files when
+  needed. The default is ``no_preprocessing``, and other options are described
+  in the :doc:`concepts/preprocessing-spec` section.
+
+- ``(preprocessor_deps (<deps-conf list>))`` specifies extra preprocessor
+  dependencies, e.g., if the preprocessor reads a generated file.
+  Dependency specification is described in the :doc:`concepts/dependency-spec`
+  section.
+
+- ``(compile_flags <flags>)`` specifies compilation flags to be passed to
+  ``melc``, the Melange compiler that produces the ``.js`` artifacts.
+  ``<flags>`` is described in detail in the
+  :doc:`reference/ordered-set-language` section. It also supports
+  ``(:include ...)`` forms. The default value for this field is taken from the
+  environment and, as a result, it's recommended to write
+  ``(compile_flags ...)`` fields as follows:
+  ``(compile_flags :standard <my options>)``
+
+- ``(root_module <module>)`` specifies a ``root_module`` that collects all
+  listed dependencies in ``libraries``. See the documentation for
+  ``root_module`` in the :ref:`library` stanza.
+
+- ``(allow_overlapping_dependencies)`` is the same as the corresponding field of
+  :ref:`library`.
+
+Libraries
+---------
+
+Melange is compatible with Dune libraries that support it. When adding support
+for Melange in libraries, two fields are involved:
+
+- ``(modes melange)``: it is required to add ``melange`` to the library
+  ``modes`` field.
+
+  - If the library has no ``modes`` field defined yet, one can use the
+    :doc:`reference/ordered-set-language` to do so: ``(modes :standard melange)``
+
+  - If the library already had some value for the ``modes`` field, ``melange``
+    mode can be appended to the list of modes, for example:
+    ``(modes byte native melange)``
+
+- ``(melange.runtime_deps <deps>)``: optionally, define any runtime dependencies
+  using ``melange.runtime_deps``. This field is analog to the ``runtime_deps``
+  field used in ``melange.emit`` stanzas. An example:
+  ``(melange.runtime_deps ./path/to/file.css (glob_file_rec ./fonts/*))``.
+
+Recommended Practices
+---------------------
+
+Keep bundles small by reducing the number of ``melange.emit`` stanzas
+=====================================================================
```

Comment:
we're not exactly consistent with heading styles, but we use `===` to indicate larger sections than `---`. Sphinx does not care but please be consistent with the order documented [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections)